### DISCUSSION TITLE
Instagram Revanced Problem

### DISCUSSION THREAD

**Discussion by fasilmrm** [ReVanced/revanced-discussions#1117] [Question]
I'm using Instagram Revanced. Whenever i tried to click and instagram link from google or whatsapp or wherever, it forwarding to a browser instead of Instagram app directly. 

So I went to phone settings > apps > default apps > opening links > instagram and i enabled everything except one which is "www.instagram.com". No matter how much i tried, I'm unable to enable toggle for this. Any solutions? Here I'm attaching the screenshot 
![Screenshot_20241213_133945_Settings](https://github.com/user-attachments/assets/7fd062e2-f873-48fd-bef9-9dbf01984f27)

https://github.com/user-attachments/assets/76a8d5d7-ea44-49d1-b099-2339b1d09f66

**Comment by LisoUseInAIKyrios** [comment]
You cannot then on the Instagram.com switch?  That makes no sense.  That switch needs to be on to open links with the app.

**Reply by fasilmrm** [reply]
Actually i can. But the moment I go out of that settings it gets disabled

https://github.com/user-attachments/assets/54772b01-6c05-4f2e-af0b-d2b1955b8240

**Reply by LisoUseInAIKyrios** [reply]
It might be Android 12 deep link verification failing because the app doesn't match the signatures the instagram.com declares.

If this is true, it seems that deep link verification can be turned off by modifying the manifest.  It also seems that it can be manually allowed using adb.

**Reply by fasilmrm** [reply]
I have no idea how to do that. My device is running on android 14

**Reply by fasilmrm** [reply]
🫤

**Reply by joolz1** [reply]
Turn off open supported links in Settings -> Apps > Meta App Manager, then it'll allow you to select www.instagram.com.

However I have done this, but Instagram just fails to open anyway..


https://github.com/user-attachments/assets/4266343a-d6db-48bb-9e95-be7fa8b7861b

**Comment by TheRiddermark** [comment]
This is still an issue. I can't open any Instagram links from any other app like Viber, Whatsapp or browser. All supported links are set to be opened by the revanced-modified instagram, but on click it results (as shown above by other posters) in instagram opening for a split second and disappearing.

**Reply by Wrakor** [reply]
Yep, same here.

**Reply by Ouroboros1337** [reply]
Same here must be a bug.
If someone wants to look at it:
1 Patch recommended Instagram
2 Set it up to open links
3 Open a Link
4 See that Instagram opens but not the picture/reel from the link

**Reply by fasilmrm** [reply]
> Same here must be a bug. If someone wants to look at it: 1 Patch recommended Instagram 2 Set it up to open links 3 Open a Link 4 See that Instagram opens but not the picture/reel from the link

It has been 3 month since i posted the issue. Seems like no one cares 🫤

**Comment by neekless** [comment]
it works if you patch instagram v350 and below. any newer versions will result in this behavior. instagram must have changed something that nobody could figure out.

**Reply by fasilmrm** [reply]
I downloaded vanced directly from here. Not patched

**Reply by neekless** [reply]
what are you talking about. they do not provide instagram revanced for download. you have to patch instagram in revanced manager

**Reply by neekless** [reply]
didn't know this existed, even though those apps are unofficial and not provided by the revanced team. if you download a v350 one it should work

**Reply by fasilmrm** [reply]
Ok

**Reply by fasilmrm** [reply]
Tnx. v350 works fine ✌🏼✌🏼

**Reply by oSumAtrIX** [reply]
This is unofficial and unsupported here

**Comment by GeCAF** [comment]
It's working for me in version 378.0.0.52.68 with this method:

> Turn off open supported links in Settings -> Apps > Meta App Manager, then it'll allow you to select [www.instagram.com](http://www.instagram.com).

by @joolz1

**Comment by UmbahaSpodPaha** [comment]
The problem is still here, no matter what I do... i even disablet Meta App Manager and nothing, links can't open any patched instagram app unless it's unmodified official .apk

Not possible to do some patch to work around it? Or some helper app which will help redirect links to patched  apks?

It looks like Instafel got it working perfectly somehow!

**Reply by joolz1** [reply]
This issue is long gone.
Working fine in the latest version.

![Screenshot_20251107_061621_Settings](https://github.com/user-attachments/assets/ca0fbc45-eb4d-4568-92b4-93030e5cb7dd)
![Screenshot_20251107_061629_Settings](https://github.com/user-attachments/assets/bab6a0a5-89c5-4965-a6ff-64225ff97207)

**Comment by BlueRocketMouse** [comment]
Is there any solution to this aside from patching an older version of Instagram? Open supported links is disabled on Meta App Manager but I still get the same behavior with Instagram opening then immediately closing.

I tried looking into disabling the link verification with adb like was mentioned earlier, but I could not find any information on how to do this.

**Reply by UmbahaSpodPaha** [reply]
I just found the first Instagram mod with newer base which opens every external link...both from browsers, messengers and stuff! FINALLY! It's called InstaPrime (clone) and it's really promising! It finally open links like original app!

**Reply by oSumAtrIX** [reply]
That's not what fixes it. The fix is in a comment to this post above this one

**Reply by UmbahaSpodPaha** [reply]
> That's not what fixes it. The fix is in a comment to this post above this one

Yeah but that fix DONT WORK for links you try to open from meta apps like Messenger and Whatsapp

**Reply by joolz1** [reply]
Working beautifully here..

**Reply by UmbahaSpodPaha** [reply]
Please tell me step by step (if you can) how to achieve Instagram Revanced to open links from messenger then because maybe I'm stupid I tried everything and opening from Messenger never worked besides original app and until I found that new mod

**Reply by joolz1** [reply]
https://github.com/orgs/ReVanced/discussions/1117#discussioncomment-11874624

**Reply by UmbahaSpodPaha** [reply]
> https://github.com/orgs/ReVanced/discussions/1117#discussioncomment-11874624

Yeah I tried exactly that many times, still not opening links clicked in Metas Messenger... Only clicks from browser worked

**Reply by BlueRocketMouse** [reply]
> https://github.com/orgs/ReVanced/discussions/1117#discussioncomment-11874624

I am confused. You say this is a fix, but this same comment shows it isn't since links still fail to open in Instagram anyway. 

In any case, I already did what this comment suggested and the issue wasn't fixed. I get the same behavior shown in the video. How did you resolve that?

**Reply by joolz1** [reply]
Download APKtool M, open your Revanced APK and "Quick Edit" and change the version code to anything other than what it is set at, i.e: use "555000555"

Also ensure you have disabled "open supported links" in Meta App Manager, and enabled all the URLs in "supported links" on your version of Revanced APKtool M modified APK

**Reply by BlueRocketMouse** [reply]
> Download APKtool M, open your Revanced APK and "Quick Edit" and change the version code to anything other than what it is set at, i.e: use "555000555"
> 
> Also ensure you have disabled "open supported links" in Meta App Manager, and enabled all the URLs in "supported links" on your version of Revanced APKtool M modified APK

It works perfectly now. Thank you!

**Reply by joolz1** [reply]
Happy to help.

Now next time you want to patch an updated version change the version code to 555000556 and so on and you can update the APK. 

Updates won't show up in google play store either 🙂

**Reply by oSumAtrIX** [reply]
If you want to change the version code there is a version code patch

**Reply by StrangePeanut** [reply]
> Download APKtool M, open your Revanced APK and "Quick Edit" and change the version code to anything other than what it is set at, i.e: use "555000555"
> 
> Also ensure you have disabled "open supported links" in Meta App Manager, and enabled all the URLs in "supported links" on your version of Revanced APKtool M modified APK

This worked brilliantly, thanks!

> If you want to change the version code there is a version code patch

Unfortunately this patch caused the build to fail instantly.

**Reply by Numeritxs** [reply]
> Download APKtool M, open your Revanced APK and "Quick Edit" and change the version code to anything other than what it is set at, i.e: use "555000555"
> 
> Also ensure you have disabled "open supported links" in Meta App Manager, and enabled all the URLs in "supported links" on your version of Revanced APKtool M modified APK

It worked! Thanks a lot

**Reply by jboe0696** [reply]
Thanks joolz1, doing what you said worked and now I can open instagram links from other apps in Instagram just fine. 
But one strange thing is that links to profiles (not reels, not posts) open the app, but don't actually do anything or go to the profile. Can anyone else reproduce? FWIW, I'm on 415.1.0.46.76 with only the 4 default patches applied, but I've also tested this on other app versions, mostly 400+; they all have the same issue. The unpatched apps, whether from apkmirror or google play, work fine on this front. I've got all the "supported links" ticked on android app settings, and the supported links turned off in Meta App Manager.

My super old versions on other devices which were patched years ago can still open profile/reel/post links, btw

**Reply by EM2Rob** [reply]
> Download APKtool M, open your Revanced APK and "Quick Edit" and change the version code to anything other than what it is set at, i.e: use "555000555"
> 
> Also ensure you have disabled "open supported links" in Meta App Manager, and enabled all the URLs in "supported links" on your version of Revanced APKtool M modified APK

When I do this and try to patch, it fails at Sign patched apk file. At a quick glance I see, Failed to determine APK's minimum supported Android platform version.
